### PR TITLE
Update packing-with-FA2.md to include SFTTrainer version

### PR DESCRIPTION
**Pull request by wynterl**
sync with current main

### CODE REVIEW

File: packing-with-FA2.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ Hugging Face Transformers now addresses this with a new feature that maintains b
 
 By selecting `DataCollatorWithFlattening`, Hugging Face `Trainer` users can now seamlessly concatenate sequences into a single tensor while accounting for sequence boundaries during Flash Attention 2 computations. This is achieved through the `flash_attn_varlen_func`, which calculates the cumulative sequence lengths in each mini-batch (`cu_seqlens`).
 
+The same feature is available to Hugging Face `SFTTrainer` users in the `TRL` library by setting a new flag `padding_free=True` when calling the data collator `DataCollatorForCompletionOnlyLM`.
```
Comment (pcuenca) [inline packing-with-FA2.md:36]:
```suggestion
The same feature is available to Hugging Face `SFTTrainer` users in the `TRL` library by setting a new flag, `padding_free=True`, when calling the data collator `DataCollatorForCompletionOnlyLM`.
```

File: packing-with-FA2.md [hunk 2]
```diff
@@ -68,19 +70,104 @@ The modifications required are lightweight and are limited to providing the `pos
...
 ## Getting started
-Reaping the benefits of packing with `position_ids` is easy. To use packing with `position_ids`, only two steps are required:
+Reaping the benefits of packing with `position_ids` is easy. 
+
+If you are using Hugging Face `Trainer` from `Transformers`, only two steps are required:
 
 1) Instantiate the model with Flash Attention 2
 2) Use the new `DataCollatorWithFlattening`
+
+If you are using Hugging Face `SFTTrainer` from `TRL` with `DataCollatorForCompletionOnlyLM`, then the two steps required are:
```
Comment (pcuenca) [inline packing-with-FA2.md:80]:
```suggestion
If you are using Hugging Face `SFTTrainer` from `TRL` with `DataCollatorForCompletionOnlyLM`, then the two required steps are:
```

File: packing-with-FA2.md [hunk 3]
```diff
@@ -68,19 +70,104 @@ The modifications required are lightweight and are limited to providing the `pos
...
+2) Set `padding_free=True`  when calling `DataCollatorForCompletionOnlyLM` as follows:
+`collator = DataCollatorForCompletionOnlyLM(response_template_ids, tokenizer=tokenizer, padding_free=True)`
    
 ## How to use it
 
-![image1](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/packing-with-FA2/image1.png)
-![image2](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/packing-with-FA2/image2.png)
+For `Trainer` users, the example below illustrates how to use the new feature. 
+
+```
```
Comment (pcuenca) [inline packing-with-FA2.md:90]:
```suggestion
```Python
```

This should trigger syntax highlight :)

File: packing-with-FA2.md [hunk 4]
```diff
@@ -68,19 +70,104 @@ The modifications required are lightweight and are limited to providing the `pos
...
+    model=model,
+    train_dataset=train_dataset,
+    data_collator=data_collator
+)
+trainer.train()
+```
+
+For `TRL` users, the example below shows how to use the new feature with `SFTTrainer`.
+
+```
```
Comment (pcuenca) [inline packing-with-FA2.md:125]:
```suggestion
```Python
```

**Review by pcuenca — APPROVED** [review]
Many thanks!

File: packing-with-FA2.md (hunk 1 above)
Comment (wynterl) [reply packing-with-FA2.md]: done

File: packing-with-FA2.md (hunk 2 above)
Comment (wynterl) [reply packing-with-FA2.md]: done

File: packing-with-FA2.md (hunk 3 above)
Comment (wynterl) [reply packing-with-FA2.md]: done

File: packing-with-FA2.md (hunk 4 above)
Comment (wynterl) [reply packing-with-FA2.md]: done

**Pull request merged**